accounts.go

```diff
@@ -66,59 +91,58 @@ func (lw *LibWallet) GetAccountBalance(accountNumber int32, requiredConfirmation
 	}, nil
 }
 
-func (lw *LibWallet) SpendableForAccount(account int32, requiredConfirmations int32) (int64, error) {
-	bals, err := lw.wallet.CalculateAccountBalance(uint32(account), requiredConfirmations)
+func (wallet *Wallet) SpendableForAccount(account int32, requiredConfirmations int32) (int64, error) {
+	bals, err := wallet.internal.CalculateAccountBalance(wallet.shutdownContext(), uint32(account), requiredConfirmations)
 	if err != nil {
 		log.Error(err)
 		return 0, translateError(err)
 	}
 	return int64(bals.Spendable), nil
 }
 
-func (lw *LibWallet) NextAccount(accountName string, privPass []byte) error {
-	_, err := lw.NextAccountRaw(accountName, privPass)
-	if err != nil {
-		log.Error(err)
-		return err
-	}
-	return nil
-}
-
-func (lw *LibWallet) NextAccountRaw(accountName string, privPass []byte) (uint32, error) {
+func (wallet *Wallet) NextAccount(accountName string, privPass []byte) (int32, error) {
 	lock := make(chan time.Time, 1)
 	defer func() {
 		for i := range privPass {
 			privPass[i] = 0
 		}
 		lock <- time.Time{} // send matters, not the value
 	}()
-	err := lw.wallet.Unlock(privPass, lock)
+
+	ctx := wallet.shutdownContext()
+	err := wallet.internal.Unlock(ctx, privPass, lock)
 	if err != nil {
 		log.Error(err)
 		return 0, errors.New(ErrInvalidPassphrase)
 	}
 
-	return lw.wallet.NextAccount(accountName)
+	accountNumber, err := wallet.internal.NextAccount(ctx, accountName)
+
+	return int32(accountNumber), err
 }
 
-func (lw *LibWallet) RenameAccount(accountNumber int32, newName string) error {
-	err := lw.wallet.RenameAccount(uint32(accountNumber), newName)
-	return err
+func (wallet *Wallet) RenameAccount(accountNumber int32, newName string) error {
+	err := wallet.internal.RenameAccount(wallet.shutdownContext(), uint32(accountNumber), newName)
+	if err != nil {
+		return translateError(err)
+	}
+
+	return nil
 }
 
-func (lw *LibWallet) AccountName(accountNumber int32) string {
-	name, err := lw.AccountNameRaw(uint32(accountNumber))
+func (wallet *Wallet) AccountName(accountNumber int32) string {
+	name, err := wallet.AccountNameRaw(uint32(accountNumber))
 	if err != nil {
 		log.Error(err)
 		return "Account not found"
 	}
 	return name
```

Comment:
For the most part of your code, you add a space after an error check or before a return statement. You should do the same here to be consistent.